src/cryptonote_core/service_node_list.cpp

```diff
@@ -1735,26 +1725,31 @@ bool service_node_list::state_t::process_confirmed_event(
 
     try {
         auto [key, service_node_info] = validate_ethereum_registration(
-                new_sn, nettype, hf_version, height, index, get_staking_requirement(nettype));
+                new_sn,
+                confirm.nettype,
+                confirm.hf_version,
+                confirm.confirmed_height,
+                confirm.vote_index,
+                get_staking_requirement(confirm.nettype));
         if (sn_list && !sn_list->m_rescanning) {
             auto& proof = sn_list->proofs[key];
             proof = {};
             proof.store(key, sn_list->blockchain);
         }
-        if (my_keys && my_keys->pub == key)
+        if (confirm.my_keys && confirm.my_keys->pub == key)
```

Comment:
It feels a little weird to have bundled `my_keys` into this struct, but it's not a big deal.